### PR TITLE
pull in jakarta ee 9 apis

### LINKED ISSUE #38: Pull in Jakarta EE 9 dependencies
In order to align with Jakarta EE 9, we need to update the dependencies on Jakarta EE 9 specs .

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Fixes: #38

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,12 +36,12 @@
...
 
         <!-- Dependencies - Jakarta -->
-        <version.jakarta.api>8.0.0</version.jakarta.api>
-        <version.jakarta.annotation>1.3.5</version.jakarta.annotation>
-        <version.jakarta.cdi>2.0.2</version.jakarta.cdi>
-        <version.jakarta.ws-rs>2.1.6</version.jakarta.ws-rs>
-        <version.jakarta.jsonp>1.1.6</version.jakarta.jsonp>
-        <version.jakarta.jsonb>1.0.2</version.jakarta.jsonb>
+        <version.jakarta.api>9.0.0</version.jakarta.api>
+        <version.jakarta.annotation>2.0</version.jakarta.annotation>
```
Comment (xstefank) [inline pom.xml]:
```suggestion
        <version.jakarta.annotation>2.0.0</version.jakarta.annotation>
```
https://search.maven.org/artifact/jakarta.annotation/jakarta.annotation-api/2.0.0/jar

Comment (xstefank) [reply pom.xml]: similarly for others, CDI should be 3.0.0, JAX-RS 3.0.0, etc.

**Review by xstefank — COMMENTED** [review]
@Emily-Jiang also if we would be adapting directly Jakarta EE 9.1 (not 9.0 as in this PR) I noticed that all versions still match except for JSON-P where there is a minor update to 2.0.1 - https://github.com/eclipse-ee4j/jakartaee-api/blob/9.1.0/pom.xml#L52. This is just a note.

File: pom.xml (hunk 1 above)
Comment (Emily-Jiang) [reply pom.xml]: basically, 2.0.0 is the same as 2.0. Anyway, I can update to reflect the link.

**Comment by Emily-Jiang** [comment]
> @Emily-Jiang also if we would be adapting directly Jakarta EE 9.1 (not 9.0 as in this PR) I noticed that all versions still match except for JSON-P where there is a minor update to 2.0.1 - https://github.com/eclipse-ee4j/jakartaee-api/blob/9.1.0/pom.xml#L52. This is just a note.

yes, Jakarta EE 9.1 is identical to Jakarta EE 9 with some old bits made optional. However the referenced APIs by MicroProfile has no changes.

**Comment by radcortez** [comment]
I think we should move the major version of the POM in this PR too (even if that can be done in the release process, it will make it clear that this is a major update).

Also, lets push a `1.x` branch with the current `master`.

File: pom.xml (hunk 1 above)
Comment (xstefank) [reply pom.xml]: No, it isn't :) -
```
[ERROR] Failed to execute goal on project dependency-test: Could not resolve dependencies for project io.xstefank:dependency-test:jar:0.0.1: Could not find artifact jakarta.annotation:jakarta.annotation-api:jar:2.0 in central (https://repo.maven.apache.org/maven2)
```

Comment (Emily-Jiang) [reply pom.xml]: ah, good point. They are the same from semantic versioning point of view :).

**Comment by Emily-Jiang** [comment]
> I think we should move the major version of the POM in this PR too (even if that can be done in the release process, it will make it clear that this is a major update).
> 
> Also, lets push a `1.x` branch with the current `master`.

When we release, we can use the version 2.0 and then the next development version will be 2.1-SNAPSHOT. We can create a 1.x branch with the tag 1.0 if needed. @radcortez thoughts?

**Comment by radcortez** [comment]
> > I think we should move the major version of the POM in this PR too (even if that can be done in the release process, it will make it clear that this is a major update).
> > Also, lets push a `1.x` branch with the current `master`.
> 
> When we release, we can use the version 2.0 and then the next development version will be 2.1-SNAPSHOT. We can create a 1.x branch with the tag 1.0 if needed. @radcortez thoughts?

I think we need the split now, because we have unreleased changes in the current `master`. So the `1.0` tag would not contain these.

**Comment by Emily-Jiang** [comment]
> > > I think we should move the major version of the POM in this PR too (even if that can be done in the release process, it will make it clear that this is a major update).
> > > Also, lets push a `1.x` branch with the current `master`.
> > 
> > 
> > When we release, we can use the version 2.0 and then the next development version will be 2.1-SNAPSHOT. We can create a 1.x branch with the tag 1.0 if needed. @radcortez thoughts?
> 
> I think we need the split now, because we have unreleased changes in the current `master`. So the `1.0` tag would not contain these.

@radcortez for the PRs you mentioned mergered in the `master`, I had a quick look and think they are okay to be released in 2.0. What kind of concerns do you have? I think we should address these by either fixing them or removing some commits if necessary?

**Comment by radcortez** [comment]
Yes, it is fine to add them to 2.0. I'm just saying it would also be easier if we do the split now (create a 1.x branch from the current master) and continue from there with 2.0.

**Comment by Emily-Jiang** [comment]
> Yes, it is fine to add them to 2.0. I'm just saying it would also be easier if we do the split now (create a 1.x branch from the current master) and continue from there with 2.0.

I see what you meant. Personally, I like to keep `master` as the main stream and open the 1.x branch for maintenance release. We can always create 1.x with the release tag easily. I would like to merge this PR into master if no further feedback and then create a 2.0-RC1 to try it out.

**Review by ederks85 — APPROVED** [review]

**Review by xstefank — APPROVED** [review]

**Comment by radcortez** [comment]
> I see what you meant. Personally, I like to keep `master` as the main stream and open the 1.x branch for maintenance release. We can always create 1.x with the release tag easily. 

That is the point. There is not tag for a 1.x version with the additional commits. You can still do it later, by picking the commit hash, but I just think it would be easier to do it now.

**Pull request merged**